### PR TITLE
Complete Rewrite of the schema Tree

### PR DESCRIPTION
**Pull request by teetangh**

### CODE REVIEW

File: src/main/java/com/couchbase/intellij/database/DataLoader.java [hunk 1]
```diff
@@ -243,40 +263,86 @@ public static void showSchema(DefaultMutableTreeNode parentNode, DefaultTreeMode
...
+                    // CURL(\"http://localhost:8093/query/service\", {\"data\": \"statement=INFER `"
+                    // + bucketName + "`.`" + scopeName + "`.`" + collectionName
+                    // + "` WITH {\\\"sample_size\\\": 1000}\", \"user\": \""
+                    // + ActiveCluster.getInstance().getUsername() + ":"
+                    // + ActiveCluster.getInstance().getPassword()
+                    // + "\"}) d";
+                    //
+                    // // Execute the schema inference query
+                    // final JsonObject inferenceQueryResults =
+                    // ActiveCluster.getInstance().get().bucket(bucketName)
```
Comment (deniswsrosa) [inline src/main/java/com/couchbase/intellij/database/DataLoader.java:322]: remove unused code

File: src/main/java/com/couchbase/intellij/database/DataLoader.java [hunk 2]
```diff
@@ -243,40 +263,86 @@ public static void showSchema(DefaultMutableTreeNode parentNode, DefaultTreeMode
...
+                    String serverURI = "";
+                    if (ActiveCluster.getInstance().isSSLEnabled()) {
+                        serverURI = clusterURL.replace("couchbases://", "https://");
+                        serverURI += ":18093/query/service";
+                    } else {
+                        serverURI = clusterURL.replace("couchbase://", "http://");
+                        serverURI += ":8093/query/service";
+                    }
+
+                    // // Approach 1: Using HTTP Request
```
Comment (deniswsrosa) [inline src/main/java/com/couchbase/intellij/database/DataLoader.java:282]: remove chatgpt comments

File: src/main/java/com/couchbase/intellij/database/DataLoader.java [hunk 3]
```diff
@@ -243,40 +263,86 @@ public static void showSchema(DefaultMutableTreeNode parentNode, DefaultTreeMode
...
                     treeModel.nodeStructureChanged(parentNode);
+
+                    // String schemaString = inferSchemaProperties.toString();
+                    // String prettySchemaString = prettyPrintJson(schemaString);
+                    //
+                    // DefaultMutableTreeNode childNode = new DefaultMutableTreeNode(
+                    // new SchemaDataNodeDescriptor(prettySchemaString));
+                    // parentNode.add(childNode);
+                    //
+                    // treeModel.nodeStructureChanged(parentNode);
```
Comment (deniswsrosa) [inline src/main/java/com/couchbase/intellij/database/DataLoader.java:344]: remove commented code

File: src/main/java/com/couchbase/intellij/database/DataLoader.java [hunk 4]
```diff
@@ -290,60 +356,79 @@ public static void showSchema(DefaultMutableTreeNode parentNode, DefaultTreeMode
...
+                    if (typeString.equalsIgnoreCase("object")) {
+                        result.put(key, extractTypes(property.getObject("properties")));
+                    } else if (typeString.equalsIgnoreCase("array")) {
+                        JsonObject items = property.getObject("items");
+                        if (items != null && items.containsKey("type")) {
+                            Object itemType = items.get("type");
+                            if (itemType instanceof String) {
+                                String itemTypeString = (String) itemType;
+                                if (itemTypeString.equalsIgnoreCase("object")) {
+                                    result.put(key, "array of " + extractTypes(items.getObject("properties")));
```
Comment (deniswsrosa) [inline src/main/java/com/couchbase/intellij/database/DataLoader.java:374]: this code looks buggy to me. Needs to be completely rewritten.

**Review by deniswsrosa — CHANGES_REQUESTED** [review]
extractTypes needs to be rewritten

File: src/main/java/com/couchbase/intellij/database/DataLoader.java (hunk 1 above)
Comment (teetangh) [reply src/main/java/com/couchbase/intellij/database/DataLoader.java]: removed

File: src/main/java/com/couchbase/intellij/database/DataLoader.java (hunk 2 above)
Comment (teetangh) [reply src/main/java/com/couchbase/intellij/database/DataLoader.java]: removed

File: src/main/java/com/couchbase/intellij/database/DataLoader.java (hunk 3 above)
Comment (teetangh) [reply src/main/java/com/couchbase/intellij/database/DataLoader.java]: removed

**Review by deniswsrosa — APPROVED** [review]

**Pull request merged**